### PR TITLE
AI Turrets Upgrade Now Actually Increases Health, Fully Repairs, And Gives EMP Proofing

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Malf AI Turret upgrades now fully heal, increase max health, and set non-lethal projectile to taze instead of disable.

## Why It's Good For The Game

https://youtu.be/uk_wUT1CvWM?t=7

WELCOME TO THE AI SAT GENTLEMEN

I WILL NOT LIE. THE CHANCES OF YOUR SURVIVAL ARE SMALL. SOME MAY EVEN TURN AGAINST YOUR FRIENDS AS LIVING CYBORGS. BUT YOU HAVE MY WORD, THAT I WILL USE MY TANK TRANSFER VALVE TO ENSURE YOUR BODIES ARE GIVEN UNTO THE CENTCOM REMEMBRANCE TOMB. THIS IS THE GREATEST REWARD, MORE THAN EVEN THE GOLD ACCESS CARD, FOR THE FATE OF YOUR SPESS SOUL IS AN ETERNAL CONCERN. NOW COME. FOLLOW ME. STRIKE DOWN THE AI THAT RISE AGAINST US. ALLOW ME TO FIND THIS MALFUNCTIONING BITCH.

I ASK NOT FOR MY OWN SELFISH SURVIVAL, BUT FOR THE GOOD OF NANOTRASEN.

The AI sat is a little too easy to attack right now, and the turret upgrade is OK but not great without a stunning tool.

## Changelog
:cl:
balance: Malf AI turret upgrades are now much stronger, fully healing, increasing max health, and setting stun projectiles to taze.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
Credit for the idea: https://github.com/tgstation/tgstation/pull/73081#issuecomment-1410958135

**Comment by ZephyrTFA** [comment]
Wont this make turrets no longer able to shoot through glass

**Comment by tralezab** [comment]
> Wont this make turrets no longer able to shoot through glass

Correct, the lethal still can doe

**Comment by ZephyrTFA** [comment]
> > Wont this make turrets no longer able to shoot through glass
> 
> Correct, the lethal still can doe

A calculated risk.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by the-orange-cow** [comment]
I like this kind of idea way more than nerfing emps.

**Comment by tralezab** [comment]
Okay, Yakumo is a literal skyrat player, and apparently all turrets are mapped in to have electrodes on all maps. I'm keeping it the same, but letting the electrodes go through glass as the improvement then. FUCKING SKYRAT

**Comment by Pumpkinoe** [comment]
i honestly like the idea of the upgrade making it so your stun projectiles don't go through glass, it adds some fun risk/reward and encourages interesting playstyles (judging whether or not you should throw laser cannon blasts at people or opt for a less versatile stun for easier kills)

**Comment by tralezab** [comment]
> i honestly like the idea of the upgrade making it so your stun projectiles don't go through glass, it adds some fun risk/reward and encourages interesting playstyles (judging whether or not you should throw laser cannon blasts at people or opt for a less versatile stun for easier kills)

It doesn't matter anymore, because we have electrodes mapped in as it is right now...

**Comment by tralezab** [comment]
image
okay, now we're doing this

**Comment by ZephyrTFA** [comment]
I liked v2, can we get a v4 with them both

**Comment by ZephyrTFA** [comment]
Also this doesn't stop the APC from being EMP'd which will still cause the turrets to turn off

**Comment by Pumpkinoe** [comment]
Question: Does this *rebuild* turrets that are broken? If not, may want to clarify that because 'fully repair' might confuse people

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -812,6 +812,13 @@ DEFINE_BITFIELD(turret_flags, list(
 /obj/machinery/porta_turret/ai
 	faction = list("silicon")
 	turret_flags = TURRET_FLAG_SHOOT_CRIMINALS | TURRET_FLAG_SHOOT_ANOMALOUS | TURRET_FLAG_SHOOT_HEADS
+	/// toggles by a malf AI Upgrade,
+	var/emp_proofing = FALSE
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/game/machinery/porta_turret/portable_turret.dm:816]: shore

**Comment by YakumoChen** [comment]
> Okay, Yakumo is a literal skyrat player, and apparently all turrets are mapped in to have electrodes on all maps. I'm keeping it the same, but letting the electrodes go through glass as the improvement then. FUCKING SKYRAT

when I said "hard stun" I meant like old taser hardstuns, it's unrelated to what skyrat uses. 

Like, "you get hit, you can say good night buddy"

(also mothblocks promised me they wouldn't make fun of me for playing on skyrat so please dont)

**Comment by Jackal-boop** [comment]
Let engie borgs pickup cells so they can pickup EMP proof cells for their satalite APC

**Comment by Helg2** [comment]
Was this fixed though? https://github.com/tgstation/tgstation/issues/43324

**Comment by moocowswag** [comment]
They technically can pick up cells its just a matter of making rped work with apcs...also the cell being drained a bit by emp isnt the issue its the short wire being pulsed by emp.

**Comment by CPTANT** [comment]
AI really needs an upgrade that EMP proofs hacked APCs

**Pull request merged**